sys/net/dsm/dsm.c

```diff
@@ -184,3 +217,32 @@ static int _find_session(sock_dtls_t *sock, sock_dtls_session_t *to_find,
     }
     return -1;
 }
+
+/*
+ * Checks if a state change is legal and changes it if that is the case.
+ * Otherwise the state will stay as it is.
+*/
+static void _set_state(dsm_state_t new_state, dsm_session_t *session_slot) {
```

Comment:
I think you need to return some value to indicate that the state change was valid and done.